compose/service.py

```diff
@@ -519,6 +522,59 @@ def pull(self, insecure_registry=False):
             )
 
 
+def get_container_data_volumes(container, volumes_option):
+    """Find the container data volumes that are in `volumes_option`, and return
+    a mapping of volume bindings for those volumes.
+    """
+    volumes = []
+
+    volumes_option = volumes_option or []
+    container_volumes = container.get('Volumes') or {}
+    original_volumes = set(volumes_option + container_volumes.keys())
+
+    for volume in original_volumes:
```

Comment:
@dnephin Support for proxies was added in Docker 1.5 via https://github.com/docker/docker/pull/9951 (docs changes for it are in https://github.com/docker/docker/pull/10192) not sure if there's another way to specify this. Given that it's just a one line change, you could try to build a 1.4.1 version with just that one change added?

(Sorry if it's a lame answer, but perhaps the links are useful :))